### DISCUSSION TITLE
When connecting to an access point using a PC or iPhone with ATOM Lite in AP mode, the message "Unable to connect to this network" appears and the access point cannot be connected.

### DISCUSSION THREAD

**Discussion by riraosan** [Hieromon/AutoConnect#529] [Q&A]
Hi. @Hieromon  Im @riraosan

> When connecting to an access point using a PC or iPhone with ATOM Lite in AP mode, the message "Unable to connect to this network" appears and the access point cannot be connected.

I would like to note in the title a brief description of the event.
I have confirmed the event of not being able to connect to the AP in the following environment.
- ATOM Lite (ESP32-PICO-D4)
- ATOM Lite was just opened.
- First connection.
- I have already set the password for that psk
- I have already set its access point name
- I used the Preferences library
- I use arduino-esp32 library 2.0.5
- I use PlatformIO IDE

Part of Source Code

```c++
  void begin(const char* SSID, const char* PASSWORD) {
    // Responder of root page and apply page handled directly from WebServer class.
    _server.on("/", [&]() {
      _content.replace("__AC_LINK__", String(AUTOCONNECT_LINK(COG_16)));
      _server.send(200, "text/html", _content);
    });

    _config.autoReconnect     = true;
    _config.reconnectInterval = 1;

    _config.psk     = "0123456789";
    _config.apid    = "ESP-G-" + String((uint32_t)(ESP.getEfuseMac() >> 32));
    _config.channel = 3;
    _config.minRSSI = 100;

    _portal.config(_config);

    bool result = false;

    printCredential();
    if (String(SSID).isEmpty() || String(PASSWORD).isEmpty()) {
      result = _portal.begin();
    } else {
      result = _portal.begin(SSID, PASSWORD);
    }

    if (result) {
      log_i("WiFi connected: %s", WiFi.localIP().toString().c_str());

      if (MDNS.begin(_hostName.c_str())) {
        MDNS.addService("http", "tcp", _httpPort);
        log_i("HTTP Server ready! Open http://%s.local/ in your browser\n", _hostName.c_str());
      } else
        log_e("Error setting up MDNS responder");
    } else {
      log_e("ESP32 can't connect to AP.");
      ESP.restart();
    }
  }
```

I used Library.

```yaml
lib_deps =
        ; Basic
        FS
        Wire
        SPI
        SPIFFS
        SD
        WiFi
        HTTPClient
        WiFiClientSecure
        Ticker
        ; for AutoConnect
        WebServer
        ESPmDNS
        DNSServer
        EEPROM
        HTTPUpdate
        Update
        Preferences
        ; from GitHub
        https://github.com/m5stack/M5GFX.git#develop
        https://github.com/Hieromon/AutoConnect.git#v1.3.7
```

The first time I access the ATOM Lite AP from my PC in the above environment, I get a "Cannot connect to this network" message and cannot get to the screen where I select the AP that ESP32 should connect to and enter the password.

Has a similar issue been committed to either? Is it known?
If not, I will commit the issue.

Thank you.

**Comment by riraosan** [comment]
Is this it?
https://github.com/Hieromon/AutoConnect/discussions/517

After a lot of research, I'm wondering if there is a problem with data storage using the Preferences library?

**Reply by riraosan** [reply]
I could not understand it, even after 8 hours of persistent research.
I think this event is fatal.
Could you please make this issue a joint assignment with you?

**Reply by Hieromon** [reply]
@riraosan Sorry for the late reply.
If you're still wrestling with this issue, I'd like to let you know a few things. But those matters may not be the cause of this problem.
- The setting value of minRSSI is RSSI; the unit of RSSI is dBm, and RSSI is negative in most cases because the received power is less than 1 mW. Your setting value; ```_config.minRSSI = 100``` is too big. (i.e., I don't think you can connect to any of the APs)
    If you see the ```XXXdBm, rejected``` message during the AutoConnect execution trace, which you can get with AC_DEBUG enabled, then ESP32 has discovered the AP, but the minRSSI setting is preventing connection attempts.
- Make sure that the dependent libraries have been selected correctly; it is recommended to refer to PlatformIO's Dependency Tree to make sure that no unintended versions have been imported.

I can spend my time to solve this problem. First you should enable the AC_DEBUG macro to get the trace. It should give us a lot of hints to resolve the problems.
